homeassistant/components/zoneminder.py

```diff
@@ -75,63 +64,17 @@ def setup(hass, config):
 
 def login():
     """Login to the ZoneMinder API."""
-    _LOGGER.debug("Attempting to login to ZoneMinder")
-
-    login_post = {'view': 'console', 'action': 'login'}
-    if ZM['username']:
-        login_post['username'] = ZM['username']
-    if ZM['password']:
-        login_post['password'] = ZM['password']
-
-    req = requests.post(ZM['url'] + '/index.php', data=login_post,
-                        verify=ZM['ssl_verification'], timeout=DEFAULT_TIMEOUT)
-
-    ZM['cookies'] = req.cookies
-
-    # Login calls returns a 200 response on both failure and success.
-    # The only way to tell if you logged in correctly is to issue an api call.
-    req = requests.get(
-        ZM['url'] + 'api/host/getVersion.json', cookies=ZM['cookies'],
-        timeout=DEFAULT_TIMEOUT, verify=ZM['ssl_verification'])
-
-    if not req.ok:
-        _LOGGER.error("Connection error logging into ZoneMinder")
-        return False
-
-    return True
-
-
-def _zm_request(method, api_url, data=None):
-    """Perform a Zoneminder request."""
-    # Since the API uses sessions that expire, sometimes we need to re-auth
-    # if the call fails.
-    for _ in range(LOGIN_RETRIES):
-        req = requests.request(
-            method, urljoin(ZM['url'], api_url), data=data,
-            cookies=ZM['cookies'], timeout=DEFAULT_TIMEOUT,
-            verify=ZM['ssl_verification'])
-
-        if not req.ok:
-            login()
-        else:
-            break
-
-    else:
-        _LOGGER.error("Unable to get API response from ZoneMinder")
-
-    try:
-        return req.json()
-    except ValueError:
-        _LOGGER.exception(
-            "JSON decode exception caught while attempting to decode: %s",
-            req.text)
+    global ZM
+    return ZM.login()
 
 
 def get_state(api_url):
     """Get a state from the ZoneMinder API service."""
-    return _zm_request('get', api_url)
+    global ZM
+    return ZM._zm_request('get', api_url)
```

Comment:
It would make more sense to make a method `get_state` in the `ZoneMinder` class. The user of the library should probably not need to know what request method to use or what api url to use, only the state id to get. Or have specific methods for different state ids.

Reply:
Agreed, this is temporary situation (so I didn't break the sensor/switch) until I was done refactoring.